scripts/build/termux_step_start_build.sh

```diff
@@ -52,11 +41,17 @@ termux_step_start_build() {
 	if [ "$TERMUX_DEBUG_BUILD" = "false" ] && [ "$TERMUX_FORCE_BUILD" = "false" ]; then
 		if [ -e "$TERMUX_BUILT_PACKAGES_DIRECTORY/$TERMUX_PKG_NAME" ] &&
 			[ "$(cat "$TERMUX_BUILT_PACKAGES_DIRECTORY/$TERMUX_PKG_NAME")" = "$TERMUX_PKG_FULLVERSION" ]; then
-			echo "$TERMUX_PKG_NAME@$TERMUX_PKG_FULLVERSION built - skipping (rm $TERMUX_BUILT_PACKAGES_DIRECTORY/$TERMUX_PKG_NAME to force rebuild)"
-			exit 0
+
+			if ! termux_validate_built_packages; then
+				echo "Force rebuilding $TERMUX_PKG_NAME@$TERMUX_PKG_FULLVERSION since some of its built package files not found"
+			else
+				echo "$TERMUX_PKG_NAME@$TERMUX_PKG_FULLVERSION built - skipping (rm $TERMUX_BUILT_PACKAGES_DIRECTORY/$TERMUX_PKG_NAME to force rebuild)"
+				exit 0
+			fi
+
 		elif [ "$TERMUX_ON_DEVICE_BUILD" = "true" ] &&
-			([[ "$TERMUX_APP_PACKAGE_MANAGER" = "apt" && "$(dpkg-query -W -f '${db:Status-Status} ${Version}\n' "$TERMUX_PKG_NAME" 2>/dev/null)" = "installed $TERMUX_PKG_FULLVERSION" ]] ||
-			 [[ "$TERMUX_APP_PACKAGE_MANAGER" = "pacman" && "$(pacman -Q $TERMUX_PKG_NAME 2>/dev/null)" = "$TERMUX_PKG_NAME $TERMUX_PKG_FULLVERSION_FOR_PACMAN" ]]); then
+			{ [[ "$TERMUX_APP_PACKAGE_MANAGER" = "apt" && "$(dpkg-query -W -f '${db:Status-Status} ${Version}\n' "$TERMUX_PKG_NAME" 2>/dev/null)" = "installed $TERMUX_PKG_FULLVERSION" ]] ||
+			 [[ "$TERMUX_APP_PACKAGE_MANAGER" = "pacman" && "$(pacman -Q $TERMUX_PKG_NAME 2>/dev/null)" = "$TERMUX_PKG_NAME $TERMUX_PKG_FULLVERSION_FOR_PACMAN" ]]; }; then
```

Comment:
```
[[ "$TERMUX_APP_PACKAGE_MANAGER" = "apt" && "$(dpkg-query -W -f '${db:Status-Status} ${Version}\n' "$TERMUX_PKG_NAME" 2>/dev/null)" = "installed $TERMUX_PKG_FULLVERSION" ]]
```
is being treated as a single expression so bash will invoke `TERMUX_APP_PACKAGE_MANAGER` variable substitution, `dpkg-query` invocation, compare values with given strings and only after that will check if both conditions are true. Probably you want to split these conditions to avoid performance hit.
Same applies to the second line.

Reply:
You might need a refresher on how `if` conditions work.

```shell
~ $ [[ 1 == 1 && "$(echo error 1>&2)" == 1 ]]
error

~ $ [[ 1 == 2 && "$(echo error 1>&2)" == 1 ]]
```